File: src/actions/remove.js
"use strict";

const flatry = require("flatry");
const mkdirp = require("make-dir");
const de = require("directory-exists");
const { paths, rmrf } = require("../utils/toolbox.js");

module.exports = async (args) => {
	let { registrypath } = paths;
	let { all } = args;
	let packages = args._;
	packages.shift();

	// Empty registry when `--all` flag is provided.
	if (all) {
		await flatry(rmrf(registrypath));
		await flatry(mkdirp(registrypath));
		packages.length = 0; // Empty array to skip loop.
	}

	// Remove provided packages.
	for (let i = 0, l = packages.length; i < l; i++) {
		let pkg = packages[i];
		let pkgpath = `${registrypath}/${pkg}`;

		let [err, res] = await flatry(de(pkgpath));
		if (err || !res) continue;
		await flatry(rmrf(pkgpath));
	}
};

File: src/parser/nim/actions/remove.nim
import std/[os, asyncdispatch, json, strformat, sequtils]

import ../utils/paths

proc nlcli_remove*(s: string = "{}") {.async.} =
    let registrypath = paths["registrypath"]

    let jdata = parseJSON(s)
    let all = jdata{"all"}.getBool()
    var packages = toSeq(jdata{"_"})
    packages.delete(0, 0)

    # Empty registry when `--all` flag is provided.
    if all:
        removeDir(registrypath)
        createDir(registrypath)
        packages.setLen(0) # Empty array to skip loop.

    # Remove provided packages.
    for pkg in packages:
        let pkgpath = fmt"{registrypath}/{pkg}"

        if dirExists(pkgpath): removeDir(pkgpath)
